### ISSUE TITLE
Implement SPAKE2+ verifier confirmation

### ISSUE THREAD

**Issue by yanesca** [Mbed-TLS/mbedtls#9352]
Depends on: https://github.com/Mbed-TLS/mbedtls/issues/9349

The task is to

- Extend `psa_pake_output()` to output `PSA_PAKE_STEP_CONFIRM` on the verifier side (PSA_KEY_TYPE_SPAKE2P_PUBLIC_KEY)
- This will include calculate the shared key and the confirmation key according to the key schedule

The goal is to make the above work with with [PSA_ALG_SPAKE2P_HMAC](https://arm-software.github.io/psa-api/crypto/1.2/ext-pake/api/pake.html#c.PSA_ALG_SPAKE2P_HMAC) for all supported [ciphersuites](https://datatracker.ietf.org/doc/html/draft-bar-cfrg-spake2plus-02#section-5)
(Edwards curves are not supported yet in Mbed TLS).

See also: the [PSA Crypto Standard](https://arm-software.github.io/psa-api/crypto/1.2/ext-pake/api/pake.html#spake2-operation) and [SPAKE2+ standard](https://datatracker.ietf.org/doc/html/rfc9383.html).

Testing:

- [test vectors](https://datatracker.ietf.org/doc/html/rfc9383.html#appendix-C) from the RFC
- tests for the rest of the ciphersuites
- negative tests

Preferably resolve this in two PRs:

1. Implementation with P256-SHA256-HKDF-HMAC with standard test vectors and negative tests
2. Further unit tests for the already covered cases and unit tests for the remaining supported ciphersuites
